library/ReplicatedTableEntity.cs

```diff
@@ -100,24 +101,112 @@ public ReplicatedTableEntity(string partitionKey, string rowKey)
+                    continue;
+                }
+
+                allProperty.SetValue(this, properties[allProperty.Name]);
+            }
         }
 
         /// <summary>
-        /// Serializes the <see cref="Dictionary{TKey,TValue}"/> of property names mapped to values of type <see cref="EntityProperty"/> from this <see cref="DynamicTableEntity"/> instance.
+        /// Serializes the <see cref="Dictionary{TKey,TValue}"/> of property names mapped to values of type <see cref="object"/> from this <see cref="DynamicTableEntity"/> instance.
         /// </summary>
-        /// <param name="operationContext">An <see cref="OperationContext"/> object used to track the execution of the operation.</param>
-        /// <returns>A collection containing the map of string property names to values of type <see cref="EntityProperty"/> stored in this <see cref="DynamicTableEntity"/> instance.</returns>
-        public virtual IDictionary<string, EntityProperty> WriteEntity(OperationContext operationContext)
+        /// <returns>A collection containing the map of string property names to values of type <see cref="object"/> stored in this <see cref="DynamicTableEntity"/> instance.</returns>
+        public virtual IDictionary<string, object> WriteEntity()
+        {
+            Dictionary<string, object> dictionary = new Dictionary<string, object>();
+            foreach (PropertyInfo allProperty in this.GetType().GetTypeInfo().GetAllProperties())
+            {
+                if (!ShouldSkipProperty(allProperty))
+                {
+                    var obj = allProperty.GetValue(this, null);
+                    if (obj != null)
+                    {
+                        dictionary.Add(allProperty.Name, obj);
+                    }
+                }
+            }
+
+            return dictionary;
+        }
+
+        public static bool ShouldSkipProperty(PropertyInfo property)
         {
-            return TableEntity.WriteUserObject(this, operationContext);
+
+            switch (property.Name)
+            {
+                case "PartitionKey":
+                case "RowKey":
+                case "Timestamp":
+                case "ETag":
+                    return true;
+                default:
+                    {
+                        MethodInfo methodInfo = property.SetMethod;
+                        MethodInfo methodInfo2 = property.SetMethod;
+                        if ((object)methodInfo == null || !methodInfo.IsPublic || (object)methodInfo2 == null || !methodInfo2.IsPublic)
+                        {
+                            ReplicatedTableLogger.LogInformational("Omitting property '{0}' from serialization/de-serialization because the property's getter/setter are not public.", property.Name);
+                            return true;
+                        }
+
+                        if (methodInfo.IsStatic)
+                        {
+                            return true;
+                        }
```

Comment:
Looks like we are not ignoring property with "IgnorePropertyAttribute" which was done in the previous library, is that intentional ?

Reply:
IgnorePropertyAttribute only exists in old package https://learn.microsoft.com/en-us/dotnet/api/microsoft.azure.cosmos.table.ignorepropertyattribute?view=azure-dotnet

I found that this issue was reported in SDK
https://github.com/Azure/azure-sdk-for-net/issues/19782

Looks like we can move to using IgnorePropertyAttribute to IgnoreDataMemberAttribute, but that also means that clients would also need to use [IgnoreDataMember] instead of [IgnoreProperty]